### PR TITLE
Intention implemented with intellij plugin

### PR DESCRIPTION
**Pull request by MuZiQiAM**

### CODE REVIEW

File: src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java [hunk 1]
```diff
@@ -41,11 +53,11 @@ final class ConditionalOperatorConverter extends PsiElementBaseIntentionAction i
      * intention menu or {@code false} for all other types of caret positions
      */
     public boolean isAvailable(@NotNull Project project, Editor editor, @Nullable PsiElement element) {
-        if (element == null) {
-            return false;
-        }
-
-        return true;
+        //element should not be null and parent should be PsiVariable
```
Comment (raphaeldichler) [inline src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java]: remove debugging comments

File: src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java [hunk 2]
```diff
@@ -88,26 +98,56 @@ public void keyPressed(KeyEvent e) {
...
-                                    element,
-                                    selectedValue,
-                                    false, // Set to true if you want to search for text occurrences
-                                    false  // Set to true if you want to rename test sources as well
-                            );
-                            renameProcessor.run();
+
+                        var pa = element.getParent();
+
+                        if (pa instanceof PsiVariable variable) {
```
Comment (raphaeldichler) [inline src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java]: can you check if you can check multiple instanceof at once

File: src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java [hunk 3]
```diff
@@ -88,26 +98,56 @@ public void keyPressed(KeyEvent e) {
...
+            WriteCommandAction.runWriteCommandAction(project, () -> {
+                try {
+                    RenameProcessor renameProcessor = new RenameProcessor(
+                            project,
+                            element,
+                            newName,
+                            true,
+                            true
+                    );
+                    // Verwende invokeLater, um den EDT nicht zu blockieren
```
Comment (raphaeldichler) [inline src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java]: remove those comments. add logging if an error happens. for logging see:     private static final Logger log = Logger.getInstance(OnStartup.class);

and use log.error/warn

File: src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java [hunk 4]
```diff
@@ -88,26 +98,56 @@ public void keyPressed(KeyEvent e) {
                     if (selectedValue != null) {
                         // todo do a replace with the suggested value
                         System.out.println("Selected: " + selectedValue);
```
Comment (raphaeldichler) [inline src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java]: please remove this comment also

File: src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java [hunk 5]
```diff
@@ -59,19 +71,16 @@ public boolean isAvailable(@NotNull Project project, Editor editor, @Nullable Ps
      * @throws IncorrectOperationException Thrown by underlying (PSI model) write action context
      *                                     when manipulation of the PSI tree fails.
      */
+    @Override
     public void invoke(@NotNull Project project, Editor editor, @NotNull PsiElement element) {
-        if (editor instanceof ImaginaryEditor) {
```
Comment (raphaeldichler) [inline src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java:64]: please revert the remove. this is a check for UI safety reason

File: src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java [hunk 6]
```diff
@@ -1,21 +1,33 @@
...
+
 import com.intellij.openapi.editor.Editor;
-import com.intellij.openapi.editor.impl.ImaginaryEditor;
+
+
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.ui.popup.JBPopup;
 import com.intellij.openapi.ui.popup.JBPopupFactory;
 import com.intellij.psi.*;
+
```
Comment (raphaeldichler) [inline src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java:17]: check formatting

File: src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java (hunk 5 above)
Comment (MuZiQiAM) [reply src/main/java/de/tum/vardoc/ConditionalOperatorConverter.java:64]: reverted

**Pull request merged**